core/storage/btree.rs

```diff
@@ -1549,7 +1676,8 @@ impl BTreeCursor {
                     .rightmost_pointer()
                 {
                     Some(right_most_pointer) => {
-                        let (mem_page, c) = self.read_page(right_most_pointer as i64)?;
+                        let (mem_page, c) =
```

Comment:
How does this work? If we yield on IO here, then `mem_page` won't be pushed to stack. If we don't yield on IO here, then is `c` ever `Some`?